src/main/java/org/springframework/samples/petclinic/rest/advice/ExceptionControllerAdvice.java

```diff
@@ -16,94 +16,124 @@
+    @ResponseBody
+    public ResponseEntity<ProblemDetail> handleRuntimeException(RuntimeException ex, WebRequest request) {
+        return createProblemDetail(
+            "Unexpected error",
+            INTERNAL_SERVER_ERROR.value(),
+            ex.getLocalizedMessage(),
+            request,
+            now()
+        );
     }
 
     /**
-     * Handles all general exceptions by returning a 500 Internal Server Error status with error details.
+     * Handles DataIntegrityViolationException and returns a {@link ProblemDetail} with HTTP 404 status.
      *
-     * @param  e The exception to be handled
-     * @return A {@link ResponseEntity} containing the error information and a 500 Internal Server Error status
+     * @param ex      the thrown DataIntegrityViolationException
+     * @param request the current WebRequest
+     * @return a ResponseEntity with a ProblemDetail and HTTP 404 status
      */
-    @ExceptionHandler(Exception.class)
-    public ResponseEntity<ErrorInfo> handleGeneralException(Exception e) {
-        ErrorInfo info = new ErrorInfo(e);
-        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(info);
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    @ResponseStatus(code = NOT_FOUND)
+    @ResponseBody
+    public ResponseEntity<?> handleDataIntegrityViolationException(DataIntegrityViolationException ex, WebRequest request) {
+        return createProblemDetail(
+            "Data Integrity Violation",
+            NOT_FOUND.value(),
+            ex.getLocalizedMessage(),
+            request,
+            now()
+        );
     }
 
     /**
-     * Handles {@link DataIntegrityViolationException} which typically indicates database constraint violations.
-     * This method returns a 404 Not Found status if an entity does not exist.
+     * Handles MethodArgumentNotValidException and returns a {@link ProblemDetail} with HTTP 400 status.
+     * Aggregates all validation error messages into a single detail string.
      *
-     * @param ex The {@link DataIntegrityViolationException} to be handled
-     * @return A {@link ResponseEntity} containing the error information and a 404 Not Found status
+     * @param ex      the thrown MethodArgumentNotValidException
+     * @param request the current WebRequest
+     * @return a ResponseEntity with a ProblemDetail and HTTP 400 status
      */
-    @ExceptionHandler(DataIntegrityViolationException.class)
-    @ResponseStatus(code = HttpStatus.NOT_FOUND)
-    @ResponseBody
-    public ResponseEntity<ErrorInfo> handleDataIntegrityViolationException(DataIntegrityViolationException ex) {
-        ErrorInfo errorInfo = new ErrorInfo(ex);
-        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorInfo);
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ProblemDetail> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex, WebRequest request) {
```

Comment:
** suggestion**: It looks like the `MethodArgumentNotValidException` had a `body` property with the `ProblemDetail `type. 
I wonder if we can reuse it then complete it?
According to its constructor, the `details` are very generic:
```
	public MethodArgumentNotValidException(MethodParameter parameter, BindingResult bindingResult) {
		super(bindingResult);
		this.parameter = parameter;
		this.body = ProblemDetail.forStatusAndDetail(getStatusCode(), "Invalid request content.");
	}
```